integration_tests/testsuite/deploy_app.py

```diff
@@ -56,21 +60,22 @@ def deploy_app(image, appdir):
             sys.exit('Error encountered when deploying app. ' +
                      'Full log: \n\n' + (output or ''))
 
-        return _retrieve_url()
+        return _retrieve_url(version)
 
     finally:
         _cleanup(appdir)
         os.chdir(owd)
 
 
 @retry(wait_fixed=10000, stop_max_attempt_number=4)
-def _retrieve_url():
+def _retrieve_url(version):
     try:
         # retrieve url of deployed app for test driver
         url_command = ['gcloud', 'app', 'describe', '--format=json']
         app_dict = json.loads(subprocess.check_output(url_command))
         hostname = app_dict.get('defaultHostname')
-        return hostname.encode('ascii', 'ignore')
+        raw_url = version + '-dot-' + hostname
```

Comment:
I think you can do:

gcloud app versions describe VERSION --service=default

and then grab the "versionUrl" field.